### PR TITLE
fix(backend): logs.py Binärmodus für garantierte Byte-Offsets (Slice K.0, Refs PR #253)

### PR DESCRIPTION
**Pull request by arn0ld87**
## Summary

Folgepatch auf den Gemini HIGH-Finding aus PR #253. `/api/logs/stream` öffnet die Logdatei jetzt im Binärmodus (`'rb'`), weil `tell()`/`seek()` im Textmodus laut Python-3-Doku opake Cookies liefern, keine garantierten Byte-Offsets. Auf Linux/macOS mit reinem LF/UTF-8 funktionierte es praktisch — aber:
- Vergleich `last_event_offset <= file_size` (st_size in Bytes) konnte inkonsistent sein
- Multi-Byte-UTF-8-Logs theoretisch verschobene IDs
- CRLF (Windows) wäre kaputt

Fix: `open('rb')` + `line.decode('utf-8', errors='replace')` pro Zeile vor Filter/Payload.

## Test plan

- [x] Neuer Test F (Multi-Byte-UTF-8 mit Umlauten + Emoji) belegt korrekte Byte-Offsets in `id:`-Frames
- [x] Bestehende 5 Tests bleiben grün
- [x] `pytest -q` Volltest: 1371 passed, 9 skipped
- [x] `ruff` + `mypy` clean

## Quelle

Gemini HIGH-Comment auf #253 (`backend/app/api/logs.py:269`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/app/api/logs.py [hunk 1]
```diff
@@ -246,18 +253,18 @@ def gen():
...
+                        # Binärmodus: tell() liefert garantiert Byte-Offsets —
+                        # im Textmodus wäre tell() ein opaker Cookie, der nicht
+                        # zwingend mit st_size oder requested_offset vergleichbar ist.
                         while True:
                             line = fh.readline()
                             if not line:
                                 break
-                            line_stripped = line.rstrip('\r\n')
                             line_offset = fh.tell()
                             offset = line_offset
```
[the remark this answers is not in the document]
Comment (arn0ld87) [reply backend/app/api/logs.py:266]: Berechtigt — fixed in PR #255 (Slice K.0.1). Geminis Snippet 1:1: `if not line.endswith(b'\n'): break` vor dem Vorrücken von `offset`. Im nächsten Poll-Zyklus wird die jetzt vollständige Zeile sauber gelesen. Neuer Test G (`vollständige Zeile mit Umlaut\nhalbe Zeile ohne newline`) belegt: exakt 1 data-Frame mit `id` nach Zeile 1, Halb-Zeile bleibt zurückgehalten. Edge-Case dokumentiert: letzte Zeile ohne `\n` (abruptes Programmende) wird nie gestreamt — akzeptabel, Logger flushen ohnehin zeilenweise.
